templates/pharmacies/general_edit.php

```diff
@@ -83,10 +83,24 @@
 function submit_pharmacy()
 {
     if(document.pharmacy.name.value.length>0)
-    {
-        top.restoreSession();
-        document.pharmacy.submit();
-        //Z&H Removed redirection
+    {   
+        //check to make sure if email is enter, it is valid
+        if(document.pharmacy.email.value.length > 0){
+            let email =  document.pharmacy.email.value;
+            let regex = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+            //console.log(document.pharmacy.email.value);
+
+            if(regex.test(email)){
+                top.restoreSession();
+                document.pharmacy.submit();
+                //Z&H Removed redirection
+            }
+
+            else{
+                alert("please enter a valid email address");
+            }
```

Comment:
Here we intend to communicate messages to the user using izi-Toast - http://izitoast.marcelodolce.com/ it is included already in the codebase see its usage here https://github.com/LibreHealthIO/lh-ehr/pull/1392/files#diff-ac491a14571b3e954dca9d6842deaeb2 and

Reply:
Also remove some trailing spaces and you are good to go